engines/order_management/spec/lib/order_management/reports/enterprise_fee_summary/report_service_spec.rb

```diff
@@ -0,0 +1,529 @@
+
+      let(:parameters_attributes) do
+        { shipping_method_ids: [shipping_method_a.id, shipping_method_b.id] }
+      end
+
+      it "filters entries" do
+        totals = service.enterprise_fee_type_totals.list
+
+        expect_total_matches(totals, 1, fee_name: "Shipping A")
+        expect_total_matches(totals, 1, fee_name: "Shipping B")
+        expect_total_matches(totals, 0, fee_name: "Shipping C")
+      end
+    end
+
+    describe "for specified payment methods" do
+      let!(:payment_method_a) do
+        create(:payment_method, name: "Payment A", distributors: [distributor])
+      end
+      let!(:payment_method_b) do
+        create(:payment_method, name: "Payment B", distributors: [distributor])
+      end
+      let!(:payment_method_c) do
+        create(:payment_method, name: "Payment C", distributors: [distributor])
+      end
+
+      let!(:order_a) { prepare_order(payment_method: payment_method_a) }
+      let!(:order_b) { prepare_order(payment_method: payment_method_b) }
+      let!(:order_c) { prepare_order(payment_method: payment_method_c) }
+
+      let(:parameters_attributes) do
+        { payment_method_ids: [payment_method_a.id, payment_method_b.id] }
+      end
+
+      it "filters entries" do
+        totals = service.enterprise_fee_type_totals.list
+
+        expect_total_matches(totals, 1, fee_name: "Payment A")
+        expect_total_matches(totals, 1, fee_name: "Payment B")
+        expect_total_matches(totals, 0, fee_name: "Payment C")
+      end
+    end
+  end
+
+  # Helper methods for example group
+
+  def expect_total_attributes(total, expected_attribute_list)
+    actual_attribute_list = [total.fee_type, total.enterprise_name, total.fee_name,
+                             total.customer_name, total.fee_placement,
+                             total.fee_calculated_on_transfer_through_name, total.tax_category_name,
+                             total.total_amount]
+    expect(actual_attribute_list).to eq(expected_attribute_list)
+  end
+
+  def expect_total_matches(totals, count, attributes)
+    expect(count_totals(totals, attributes)).to eq(count)
+  end
+
+  def prepare_tax_category(name)
+    create(:tax_category, name: name)
+  end
+
+  def default_order_options
+    { customer: customer, distributor: distributor, order_cycle: order_cycle,
+      shipping_method: shipping_method, variant: variant }
+  end
+
+  def setup_order(options = {})
+    target = default_order_options.merge(options)
+
+    create(:order, customer: target[:customer], distributor: target[:distributor],
+                   order_cycle: target[:order_cycle],
+                   shipping_method: target[:shipping_method]).tap do |order|
+      create(:line_item, order: order, variant: target[:variant])
+      order.reload
+    end
+  end
+
+  def prepare_order(options = {})
+    order = setup_order(options)
+    complete_order(order, options)
+    order.reload
+  end
```

Comment:
I think this is acceptable but I believe it would be better to use factories for this, like: https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/factories.rb#L298

I think it's better because by putting it in factories you would make the code re-usable.

Reply:
I will do this in a separate PR too, because I'd need to better test the state of the orders when there's a need to label. :slightly_smiling_face: It was trickier to move these methods to a factory, in the first few versions of the file, but it seems doable now.